### PR TITLE
Http Param Style Types

### PR DESCRIPTION
**Pull request by pytlesk4**
- add HttpParamStyles enum
- add style to IHttpParam as optional parameter
- type style on IHttpEncoding

### CODE REVIEW

File: http.d.ts [hunk 1]
```diff
@@ -34,29 +35,54 @@ export interface IHttpParam {
...
+  style?: HttpParamStyles.form;
 }
 
 export interface IHttpEncoding {
   property: string;
   mediaType?: string;
   headers?: IHttpHeaderParam[];
-  style?: 'form' | 'spaceDelimited' | 'pipeDelimited' | 'deepObject'; // defaults to form
+  // deafults to form
+  style:
```
[the remark this answers is not in the document]
Comment (marbemac) [reply http.d.ts]: typings are not responsible for default values (can't define that), the implementation is

File: http.d.ts [hunk 2]
```diff
@@ -34,29 +35,54 @@ export interface IHttpParam {
...
 }
 
+declare const enum HttpParamStyles {
+  simple = 'simple',
+  matrix = 'matrix',
+  label = 'label',
+  form = 'form',
+  spaceDelimited = 'spaceDelimited',
+  pipeDelimited = 'pipeDelimited',
+  deepObject = 'deepObject',
```
Comment (marbemac) [inline http.d.ts]: In general the TS world uses `PascalCase` for enum names - should we follow that?

Comment (pytlesk4) [reply http.d.ts]: I am good with that!

Comment (pytlesk4) [reply http.d.ts]: resolved

**Pull request merged**